tests/agents/test_write_article_image_handling.py

```
@@ -0,0 +1,107 @@
+from typing import Any
+
+import pytest
+
+from app.agents.streaming.handler import StreamEventHandler
+from app.agents.unified.tools import write_article_tool
+from app.agents.unified import tools as unified_tools
+
+
+@pytest.mark.asyncio
+async def test_write_article_rewrites_non_image_markdown_urls(monkeypatch: pytest.MonkeyPatch) -> None:
+    async def _fake_validate(url: str) -> tuple[bool, str | None, str, str]:
```

Ruff (0.15.0)

[warning] Unused function argument: `url`

(ARG001)